tests/testthat/test-num_equal.R

```diff
@@ -30,8 +30,9 @@ test_that("infinite values are handled properly", {
   expect_equal(num_equal(-Inf, Inf, tolerance = 1.e-8), FALSE)
 })
 
-test_that("NaN is different from NA_real_", {
+test_that("NaN is different from NA_real_ when tolerance set", {
   expect_false(num_equal(NaN, NA_real_), FALSE)
+  expect_true(num_equal(NaN, NA_real_, tolerance = NULL), FALSE)
```

Comment:
```suggestion
test_that("NaN is equal to NA_real_ unless tolerance is NULL", {
  expect_true(num_equal(NaN, NA_real_))
  expect_false(num_equal(NaN, NA_real_, tolerance = NULL))
```

Requires the `!` removal from above